packages/core-flows/src/definition/cart/steps/get-variant-price-sets.ts

```diff
@@ -0,0 +1,65 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { IPricingModuleService, RegionDTO } from "@medusajs/types"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+interface StepInput {
+  variantIds: string[]
+  // TODO: Expand the context passed to calculatePrices
+  region: RegionDTO
+}
+
+export const getVariantPriceSetsStepId = "get-variant-price-sets"
+export const getVariantPriceSetsStep = createStep(
+  getVariantPriceSetsStepId,
+  async (data: StepInput, { container }) => {
+    if (!data.variantIds.length) {
+      return new StepResponse({})
+    }
+
+    const pricingModuleService = container.resolve<IPricingModuleService>(
+      ModuleRegistrationName.PRICING
+    )
+
+    const variables = {
+      variant_id: data.variantIds,
+      take: null, // TODO: We probably want to add a limit here
```

Comment:
we added `null` on the take here specifically to get all variants, since by default the limit is 15 and we miss out calculating some variant prices.

Other option is to paginate them until the last item, which we haven't done.